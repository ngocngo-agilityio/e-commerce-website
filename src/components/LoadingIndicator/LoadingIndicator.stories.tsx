import type { Meta, StoryObj } from '@storybook/react';

// Components
import LoadingIndicator from '.';

const meta = {
  title: 'LoadingIndicator',
  component: LoadingIndicator,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof LoadingIndicator>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
