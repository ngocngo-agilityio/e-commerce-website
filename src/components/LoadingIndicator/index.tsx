// Libs
import { Box, Spinner as SpinnerChakraUI } from '@chakra-ui/react';

const LoadingIndicator = (): JSX.Element => {
  return (
    <Box
      style={{
        position: 'absolute',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'loadingIndicatorBg',
        zIndex: 9999,
      }}
    >
      <SpinnerChakraUI
        size="lg"
        thickness="4px"
        speed="0.65s"
        color="blue.500"
      />
    </Box>
  );
};

export default LoadingIndicator;
