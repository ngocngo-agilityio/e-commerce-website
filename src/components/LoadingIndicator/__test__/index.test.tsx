// Libs
import { render } from '@testing-library/react';

// Components
import LoadingIndicator from '..';

describe('LoadingIndicator component', () => {
  test('should render LoadingIndicator successfully', () => {
    const { container } = render(<LoadingIndicator />);

    expect(container).toBeInTheDocument();
  });

  test('should match snapshot for LoadingIndicator', () => {
    const { container } = render(<LoadingIndicator />);

    expect(container).toMatchSnapshot();
  });
});
