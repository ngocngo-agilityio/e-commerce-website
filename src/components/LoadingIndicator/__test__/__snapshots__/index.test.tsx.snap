// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator component should match snapshot for LoadingIndicator 1`] = `
<div>
  <div
    class="css-0"
    style="position: fixed; top: 0px; left: 0px; width: 100%; height: 100%; display: flex; align-items: center; justify-content: center; background-color: black; opacity: 0.7; z-index: 9999;"
  >
    <div
      class="chakra-spinner css-gqn43r"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
